### ISSUE TITLE
Hangs on "Waiting for time sync (systemd-timesyncd.service) to complete." during install

### ISSUE THREAD

**Issue by peracchi** [archlinux/archinstall#2144]
As the title says, `archinstall` hangs on _"Waiting for time sync (systemd-timesyncd.service) to complete."_ during install IF NETWORK DO NOT ALLOW NTP TRAFFIC TO GO OUT.

It do not timeout as expected.

The only way I found to do my installation was to open another SSH session and:
```
systemctl stop systemd-timesyncd
nano /etc/systemd/timesyncd.conf
<edit file>
NTP=<my internal NTP server>
<save file>
systemctl start systemd-timesyncd
```

Then, installation proceeds.

**Comment by Torxed** [comment]
Can you confirm that this is happening because network configuration is done manually (like wifi, slow DHCP or even static IP)?

**Comment by peracchi** [comment]
Yes, at work we have a very restrictive firewall, MITM, self-signed certificates and other annoyances.

One or more of these factor prevent NTP communication with outside servers and things only work after I stop NTP service, point it to internal NTP server and then restart NTP service.

**Comment by Torxed** [comment]
I figured something like this was going on ^^
Without even checking, I'm pretty sure this happens:

1. Boot
2. timesync service tries to sync
3. It fails and restart mechanics sucks
4. You configure network
5. Run archinstall
6. It waits for the next timesync to trigger
7. You SSH because the sync never happens
8. You fix it

This is an issue for manual installs to, and I think instead of fixing it here, I'll discuss with some other upstream devs if we can't solve this permanently for all use cases.

**Comment by peracchi** [comment]
I'm installing Arch on a VM on Proxmox using the ISO `archlinux-2023.09.01-x86_64.iso`

There are several other steps I need to do before starting the actual installation.

These steps are:
- boot ISO on Proxmox VM
- change root password to SCP/SSH
- SCP internal self-signed certificates
- `trust anchor <certificates>`
- `pacman-key --init && pacman-key --populate archlinux && pacman -Syy`
- `pacman -S --noconfirm archinstall`
- only then `archinstall`

**Comment by peracchi** [comment]
> We could however add a `--skip-ntp` to the command line.

This would be sufficient.

But the ideal would be to detect the problem with NTP (by timeout, perhaps?), kill the NTP process and proceed with the installation.

**Comment by alan-barzilay** [comment]
> But the ideal would be to detect the problem with NTP (by timeout, perhaps?), kill the NTP process and proceed with the installation.

Some sort of feedback to the user would also be good. When I got this issue I simply waited to see if it it would magically fix itself haha A disclaimer with possible reasons for hanging would give the user enough information to possibly do something about it himself

**Comment by Torxed** [comment]
Those are valid suggestions,

> kill the NTP process and proceed with the installation.

This is usually not enough tho, as it also requires manual configuration first.

I'll make sure to add the parameter before release and perhaps a note in the `while` loop waiting for the time sync referencing the docs relating to the issue:

![screenshot](https://github.com/archlinux/archinstall/assets/861439/e53055c5-3719-4436-8e03-50434327d28c)

**Comment by peracchi** [comment]
Thank you for your work/effort.

**Comment by alan-barzilay** [comment]
out of curiosity, is the ntp process important for any other step in the install? because I have an ansible playbook that later on configures ntp among other things, if I used the proposed '--skip-ntp' flag would I need to do anything else that was skipped/affected during archinstall?

**Comment by Torxed** [comment]
Yes, the main reason for making sure NTP exists before `archinstall` does anything is because numerous HTTPS calls, one obvious and big one is the `pacman` call which will contact HTTPS mirrors at some point. The less obvious is the `archlinux-keyring` update service that happens on the ISO in the background.

So NTP is a cheap reliable:tm: way of ensuring that such things work.

But as long as you ensure time is accurate to within a couple of seconds from the mirrors etc, any method of setting the correct time should work in theory.

**Comment by alan-barzilay** [comment]
I see, thank you for the clarification. After a quick search, ntp seems to be ill suited for VMs if they are not constantly running, [virtualbox even marked it as wontfix](https://www.virtualbox.org/ticket/3569). Letting the user skip ntp and synchronize time on his owns seems more and more like a solid solution (or add something like a vm-mode to synchronize against host time, but no idea how to do that consistently from inside guest for all virtualization providers)

**Comment by Torxed** [comment]
> I see, thank you for the clarification. After a quick search, ntp seems to be ill suited for VMs if they are not constantly running, [virtualbox even marked it as wontfix](https://www.virtualbox.org/ticket/3569). Letting the user skip ntp and synchronize time on his owns seems more and more like a solid solution (or add something like a vm-mode to synchronize against host time, but no idea how to do that consistently from inside guest for all virtualization providers)

Correct. The issue is that most often, you'll need to configure for instance [-rtc](https://www.qemu.org/docs/master/system/invocation.html) needs to be configured before launching a virtual machine and there's not much we can do after-the-fact. And as you eluded to, some virtualization platforms have issues retaining time.
So yea, a parameter might be the only viable option.

I just.. \*grinds my teeth*.. foresee the following issue being that the keyring is out of date, followed by `pacman` calls failing "arbitrarily", followed by some other network related service breaking hehe. Either way, </rant> and we'll take it as it comes. The docs now reference those issues that we can foresee.

**Comment by Torxed** [comment]
For what it's worth, a test page is published for the new docs and it references this issue and it's workaround: https://archlinux.github.io/archinstall/help/known_issues.html#waiting-for-time-sync-2144

Together with PR #2225 - adding `--skip-ntp` I think we're done :)
I'll close this once the PR is in and the docs is official.

**Comment by Torxed** [comment]
parameter is now in `master` and should make it to the next release and ISO :)

**Comment by palaboy** [comment]
another error:
![archinstall error2](https://github.com/archlinux/archinstall/assets/2505495/80fa1573-752a-4669-99f9-37be98766f92)

**Comment by svartkanin** [comment]
@Torxed might be worth putting that into the FAQ of the readme as well

**Comment by Torxed** [comment]
Good idea, we should also swap the link to the documentation to https://archinstall.archlinux.page/!

**Comment by kRHYME7** [comment]
For noobs like me, following the discussions above and can't figure it  out, just find the ntp server that best suites you. 

for me setting  ` NTP=time.google.com ` in ` /etc/systemd/timesyncd.conf ` works. 

https://www.reddit.com/r/archlinux/comments/15zgd4y/time_sync_wont_sync_help/

**Comment by Furina-cell** [comment]
> nano /etc/systemd/timesyncd.conf
> <edit file>
> NTP=<my internal NTP server>
> <save file>


what if dont't even have an NTP server?

i even have the NTP option disabled in the installation but it still gets stuck on the time sync not completing

**Comment by Stradios** [comment]
# 🛠️ Fix: `archinstall` Hangs on "Waiting for time sync"

## ❗ Problem

During an Arch Linux installation using `archinstall`, the process can hang indefinitely at:

```
Waiting for time sync (systemd-timesyncd.service) to complete.
```

This typically occurs if your **network blocks NTP traffic**, preventing time synchronization.

---

## ✅ Solution

You can manually configure `systemd-timesyncd` to bypass the NTP issue.

Reference: [Arch Wiki – systemd-timesyncd](https://wiki.archlinux.org/title/Systemd-timesyncd)

---

## 📝 Modify Time Sync Configuration

### Stop the service:

```bash
systemctl stop systemd-timesyncd
```

### Edit the config:

```bash
nano /etc/systemd/timesyncd.conf
```

Add or replace with:

```ini
[Time]
NTP=0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org
FallbackNTP=0.pool.ntp.org 1.pool.ntp.org 0.fr.pool.ntp.org
```

Exit and save (`Ctrl + X`, then press `Y` and `Enter`).

### Start the service:

```bash
systemctl start systemd-timesyncd
```

---

## 🎉 Done!

Now your system should sync time properly. You can continue the installation:

```bash
archinstall
```

---

## 🧠 Notes

- This workaround is especially helpful in corporate or school networks that block outbound NTP (UDP port 123).
- You can use a local NTP server if available.

---


if you need a more detailed fix go to.
https://github.com/Stradios/-Fix-archinstall-Hangs-on-Waiting-for-time-sync-

**Comment by Torxed** [comment]
> https://wiki.archlinux.org/title/Systemd-timesyncd
> 
> /etc/systemd/timesyncd.conf or /etc/systemd/timesyncd.conf.d/local.conf [Time] NTP=0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org FallbackNTP=0.pool.ntp.org 1.pool.ntp.org 0.fr.pool.ntp.org
> 
> solution systemctl stop systemd-timesyncd nano /etc/systemd/timesyncd.conf NTP=0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org FallbackNTP=0.pool.ntp.org 1.pool.ntp.org 0.fr.pool.ntp.org systemctl start systemd-timesyncd
> 
> after run archinstall

What am I missing here, the two texts are identical?
```
NTP=0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org
FallbackNTP=0.pool.ntp.org 1.pool.ntp.org 0.fr.pool.ntp.org
```

**Comment by Stradios** [comment]
> > https://wiki.archlinux.org/title/Systemd-timesyncd
> > /etc/systemd/timesyncd.conf or /etc/systemd/timesyncd.conf.d/local.conf [Time] NTP=0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org FallbackNTP=0.pool.ntp.org 1.pool.ntp.org 0.fr.pool.ntp.org
> > solution systemctl stop systemd-timesyncd nano /etc/systemd/timesyncd.conf NTP=0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org FallbackNTP=0.pool.ntp.org 1.pool.ntp.org 0.fr.pool.ntp.org systemctl start systemd-timesyncd
> > after run archinstall
> 
> What am I missing here, the two texts are identical?
> 
> ```
> NTP=0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org
> FallbackNTP=0.pool.ntp.org 1.pool.ntp.org 0.fr.pool.ntp.org
> ```

yes sory i just paste it form the arch linux wiki  i can edit it

**Comment by Stradios** [comment]
> > https://wiki.archlinux.org/title/Systemd-timesyncd
> > /etc/systemd/timesyncd.conf or /etc/systemd/timesyncd.conf.d/local.conf [Time] NTP=0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org FallbackNTP=0.pool.ntp.org 1.pool.ntp.org 0.fr.pool.ntp.org
> > solution systemctl stop systemd-timesyncd nano /etc/systemd/timesyncd.conf NTP=0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org FallbackNTP=0.pool.ntp.org 1.pool.ntp.org 0.fr.pool.ntp.org systemctl start systemd-timesyncd
> > after run archinstall
> 
> What am I missing here, the two texts are identical?
> 
> ```
> NTP=0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org
> FallbackNTP=0.pool.ntp.org 1.pool.ntp.org 0.fr.pool.ntp.org
> ```

https://github.com/archlinux/archinstall/issues/2144#issuecomment-3057067334 
Is this better?
